crates/iron-remote-desktop/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+mod clipboard;
+mod cursor;
+mod error;
+mod input;
+mod session;
+
+pub use clipboard::{ClipboardContent, ClipboardTransaction};
+pub use cursor::CursorStyle;
+pub use error::{IronError, IronErrorKind};
+pub use input::{DeviceEvent, InputTransaction};
+pub use session::{Session, SessionBuilder, SessionTerminationInfo};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[wasm_bindgen]
+#[derive(Clone, Copy)]
+pub struct DesktopSize {
+    pub width: u16,
+    pub height: u16,
+}
+
+#[wasm_bindgen]
+impl DesktopSize {
+    pub fn init(width: u16, height: u16) -> Self {
+        DesktopSize { width, height }
+    }
+}
+
+pub fn iron_init(log_level: &str) {
+    // When the `console_error_panic_hook` feature is enabled, we can call the
+    // `set_panic_hook` function at least once during initialization, and then
+    // we will get better error messages if our code ever panics.
+    //
+    // For more details see
+    // https://github.com/rustwasm/console_error_panic_hook#readme
+    #[cfg(feature = "panic_hook")]
+    console_error_panic_hook::set_once();
+
+    if let Ok(level) = log_level.parse::<tracing::Level>() {
+        set_logger_once(level);
+    }
+}
+
+fn set_logger_once(level: tracing::Level) {
+    use tracing_subscriber::filter::LevelFilter;
+    use tracing_subscriber::fmt::time::UtcTime;
+    use tracing_subscriber::prelude::*;
+    use tracing_web::MakeConsoleWriter;
+
+    static INIT: std::sync::Once = std::sync::Once::new();
+
+    INIT.call_once(|| {
+        let fmt_layer = tracing_subscriber::fmt::layer()
+            .with_ansi(false)
+            .with_timer(UtcTime::rfc_3339()) // std::time is not available in browsers
+            .with_writer(MakeConsoleWriter);
+
+        let level_filter = LevelFilter::from_level(level);
+
+        tracing_subscriber::registry().with(fmt_layer).with(level_filter).init();
+    })
+}
+
+#[macro_export]
+macro_rules! export_wasm {
+    ($iron_init_fn:path, $ty_session:ty, $ty_session_builder:ty, $ty_session_termination_info:ty, $ty_device_event:ty, $ty_input_transaction:ty, $ty_clipboard_transaction:ty, $ty_clipboard_content:ty, $ty_error:ty) => {
+        mod __wasm_ffi {
+            use wasm_bindgen::prelude::*;
+            use web_sys::{js_sys, HtmlCanvasElement};
+            use $crate::{
+                ClipboardContent as _, ClipboardTransaction as _, DeviceEvent as _, InputTransaction as _,
+                IronError as _, Session as _, SessionBuilder as _, SessionTerminationInfo as _,
+            };
+
+            #[wasm_bindgen]
+            pub fn iron_init(log_level: &str) {
+                $iron_init_fn(log_level);
+            }
+
```

Comment:
suggestion: So we don’t have to call manually the `iron_init` function.

```suggestion
            #[wasm_bindgen]
            pub fn iron_init(log_level: &str) {
                $crate::iron_init(log_level);
                $iron_init_fn();
            }
```